### ISSUE TITLE
[New feature] Better to add earlystopping metrics

### ISSUE THREAD

**Issue by Teng-Martin-Ma** [MDIL-SNU/SIMPLE-NN_v2#99]
If I was not missing, SIMPLE-NN currently doesn't deploy the early stopping metrics. Since it leans on PyTorch, ReduceLROnPlateau class should be a ready solution. After all, people don't want it runs forever (predefined a large max_epoch). 

Thanks for your effort in developing this package, excellent work :)

**Comment by JisuJung928** [comment]
Dear Teng-Martin-Ma,

Really sorry for the late reply.
In terms of the early stopping metrics, SIMPLE-NN already provides them as "energy_criteria".
Please read this page. (https://simple-nn-v2.readthedocs.io/en/latest/inputs/input.yaml/neural_network/energy_criteria.html)
Is it the one you want?

Best regards,
Jisu.

**Comment by Teng-Martin-Ma** [comment]
Hi Jisu,

Thanks for reply. I mean the metric monitoring the loss such as validation loss to make the early-stopping, not just the target energy or forces values. Because the model would never achieve your predefined values if the target values are so small, and we want it stop in advance if validation loss never improve in many steps, say threshold.

It is not a serious problem, but may somehow help SIMPLE-NN more reasonable when training. Have a good day!

Best regards,
Teng

-----------------------------------------
Dept. of Chemical Engineering
University College London
On Oct 23, 2022, 3:17 PM +0100, JisuJung928 ***@***.***>, wrote:

Dear Teng-Martin-Ma,

Really sorry for the late reply.
In terms of the early stopping metrics, SIMPLE-NN already provides them as "energy_criteria".
Please read this page. (https://simple-nn-v2.readthedocs.io/en/latest/inputs/input.yaml/neural_network/energy_criteria.html)
Is it the one you want?

Best regards,
Jisu.
